services/addons/images/firmware_manager/commsignia_upgrader.py

```diff
@@ -60,6 +69,61 @@ def upgrade(self):
             self.cleanup()
             self.notify_firmware_manager(success=False)
 
+    def post_upgrade(self):
+        if self.wait_until_online() == -1:
+            raise Exception("RSU offline for too long after firmware upgrade")
+        try:
+            time.sleep(60)
+            # Make connection with the target device
+            logging.info("Making SSH connection with the device...")
+            ssh = SSHClient()
+            ssh.set_missing_host_key_policy(WarningPolicy)
+            ssh.connect(
+                self.rsu_ip,
+                username=self.ssh_username,
+                password=self.ssh_password,
+                look_for_keys=False,
+                allow_agent=False,
+            )
+
+            # Make SCP client to copy over the post upgrade script to the /tmp/ directory on the remote device
+            logging.info("Copying post upgrade script to the device...")
+            scp = SCPClient(ssh.get_transport())
+            scp.put(self.post_upgrade_file_name, remote_path="/tmp/")
+            scp.close()
+
+            # Change permissions and execute post upgrade script
+            logging.info("Running post upgrade script...")
+            ssh.exec_command(
+                f"chmod +x /tmp/post_upgrade.sh"
+            )
+            _stdin, _stdout, _stderr = ssh.exec_command(
+                f"/tmp/post_upgrade.sh"
+            )
+            decoded_stdout = _stdout.read().decode()
+            logging.info(decoded_stdout)
+            if "ALL OK" not in decoded_stdout:
+                ssh.close()
+                logging.error(f"Failed to execute post upgrade script for rsu {self.rsu_ip}: {decoded_stdout}")
+                return
+            ssh.close()
+            logging.info(f"Post upgrade script executed successfully for rsu: {self.rsu_ip}.")
+        except Exception as err:
+            logging.error(f"Failed to execute post upgrade script for rsu {self.rsu_ip}: {err}")
+            
+    def wait_until_online(self):

```

Comment:
Since this function is used by both Commsignia and Yunex upgrade scripts, lets make this function a common one and move it to the upgrader.py

Reply:
Pulled this out to upgrader.py and updated unit tests to account for this change